src/bsp/src/posix/src/SocketContainer.cpp

```diff
@@ -3,12 +3,15 @@
 #include "TCPClient.h"
 #include "bsp/SettingsContainer.h"
 #include "logger/LoggerContainer.h"
+#include "LockGuard.h"
 
 std::optional<TCPClientWrapper> SocketContainer::getHostClientSocket() {
 
     static std::optional<TCPClient> s_clientSocket = {};
     static std::optional<TCPClientWrapper> s_wrapper = {};
+    static Mutex s_mutex = Mutex(500);
```

Comment:
I guess 500 is the delay to acquire the mutex with freertos. Could we reduce this delay? 500 seems a bit long and at this length, it should be either blocking or a lower timeout.

Reply:
I left a longer delay because the mutex locks the socket ::connect which is a long operation